content/recipes/necord.md

```diff
@@ -0,0 +1,365 @@
+| `StringOption`      | `string`                      | A string option      |
+| `NumberOption`      | `number`                      | A number option      |
+| `IntegerOption`     | `number`                      | An integer option    |
+| `BooleanOption`     | `boolean`                     | A boolean option     |
+| `UserOption`        | `User`                        | A user option        |
+| `MemberOption`      | `GuildMember`                 | A member option      |
+| `ChannelOption`     | `GuildChannel`                | A channel option     |
+| `RoleOption`        | `Role`                        | A role option        |
+| `MentionableOption` | `GuildMember \| Role \| User` | A mentionable option |
+| `AttachmentOption`  | `AttachmentOption`            | An attachment option |
+
+##### Autocomplete
+
+To add autocomplete to your Slashcommand you will need a interceptor first. This class will intercept all requests from the user after typing in the autocomplete option field.
+
+```typescript
+@@filename(anime.interceptor)
+import { Injectable, UseInterceptors } from '@nestjs/common';
+import { AutocompleteInteraction, CommandInteraction } from 'discord.js';
+import { AutocompleteInterceptor } from 'necord';
+
+@Injectable()
+class AnimeAutocompleteInterceptor extends AutocompleteInterceptor {
+    public transformOptions(interaction: AutocompleteInteraction) {
+        const focused = interaction.options.getFocused(true);
+        let choices: string[];
+
+        if (focused.name === 'anime') {
+            choices = ['Hunter x Hunter', 'Naruto', 'One Piece'];
+        }
+
+        return interaction.respond(
+            choices
+                .filter(choice => choice.startsWith(focused.value.toString()))
+                .map(choice => ({ name: choice, value: choice }))
+        );
+    }
+}
+```
+
+You'll then have to add `autocomplete: true` to your options class:
+
+```typescript
+@@filename(anime.dto)
+import { StringOption } from 'necord';
+
+export class AnimeDto {
+    @StringOption({
+        name: 'anime',
+        description: 'The anime to look up',
+        autocomplete: true,
+        required: true
+    })
+    anime: string;
+}
+```
+
+And last but not least, apply the interceptor to your slash command:
+
+```typescript
+@@filename(anime.commands)
+import { Injectable, UseInterceptors } from '@nestjs/common';
+import { Context, SlashCommand, Options, SlashCommandContext } from 'necord';
+import { AnimeDto } from './anime.dto';
+import { AnimeAutocompleteInterceptor } from './anime.interceptor';
+
+@Injectable()
+export class AnimeCommands {
+    @UseInterceptors(AnimeAutocompleteInterceptor)
+    @SlashCommand({
+        name: 'anime',
+        description: 'Lookup information about an anime'
+    })
+    public async onSearch(@Context() [interaction]: SlashCommandContext, @Options() { anime }: AnimeDto) {
+        return interaction.reply({content: `I found the anime ${anime}`});
+    }
+}
+```
```

Comment:
Personally, I love this! It's awesome! But to keep it more inline with the rest of our docs, do you think you could make it a `CatsAutocompleteInterceptor` that checks for cat breeds and adjust the command and descriptions accordingly?

Reply:
@jmcdo29 Yes! Of course, I will make changes within a few hours, already reverted the spaces

Reply:
@jmcdo29 Updated autocomplete category at [13f2ea3](https://github.com/nestjs/docs.nestjs.com/pull/2656/commits/13f2ea379a501b9e7540f12913ec87f008f7c05f)